services/core/events/blockchain.js

```diff
@@ -43,17 +46,11 @@ module.exports = [
 		name: 'transactions.confirmed',
 		description: 'Keep confirmed transaction list up-to-date',
 		controller: callback => {
-			coreSocket.socket.on('blocks/change', async data => {
-				logger.debug('Scheduling block list reload...');
-				const emitData = await core.getBlocks({ blockId: data.id });
-
-				if (Array.isArray(emitData.data) && emitData.data.length > 0
-					&& emitData.data[0].numberOfTransactions > 0) {
-						const transactionData = await core.getTransactions({ blockId: data.id });
-						recentBlocksCache.addNewBlock(emitData.data[0], transactionData);
-						callback(transactionData);
-				} else {
-					recentBlocksCache.addNewBlock(emitData.data[0], []);
+			signals.blockCached.add(async (block) => {
+				if (block.numberOfTransactions > 0) {
+					logger.debug(`New transactions arrived (${block.id}), ${block.numberOfTransactions} txs...`);
```

Comment:
```suggestion
					logger.debug(`${block.numberOfTransactions} new transactions arrived for block: ${block.id}`);
```

Reply:
I did not want to start the sentence with a number, and move those numbers as close to each other as possible, having that value after the blockId. Can you provide any other solution?

Reply:
Updated the suggestion.